### PR TITLE
feat: redesign farms list - add farm page

### LINKED ISSUE #73: Redesign farming page
The farming page was built as a stop gap arrangement. We need to have a look into it and see if we can make it better for desktop in terms of design.
And its totally broken in mobile, so we need a new design for how we could condense all that data into mobile viewport.

### PR DESCRIPTION
**Pull request by guerrap**
Fixes #73 

Done:
- [x] Redesign farms list
- [x] Handle token fetching error
- [x] Fix farms list mobile view
- [x] Fix failing blocks queries

image
image

### CODE REVIEW

**Comment by guerrap** [comment]
Add single farming campaign page:

image

File: src/components/FarmingList/ListItem.jsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  const below680 = useMedia('(max-width: 680px)');
+  const below1080 = useMedia('(max-width: 1080px)');
+
+  const nativeCurrency = useNativeCurrencySymbol();
+  const nativeCurrencyWrapper = useNativeCurrencyWrapper();
+  const selectedNetwork = useSelectedNetwork();
+
+  const carrotLogo = `${window.location.origin}${
+    carrotListLogoUrl.startsWith('.') ? carrotListLogoUrl.substring(1) : carrotListLogoUrl
+  }`;
```
Comment (wixzi) [inline src/components/FarmingList/ListItem.jsx:82]: Is it required?

**Comment by wixzi** [comment]
**Can you fix the unwanted space before title. in mobile view?**

Screenshot 2022-09-23 at 09 54 58


Screenshot 2022-09-23 at 09 55 06

**Comment by guerrap** [comment]
> **Can you fix the unwanted space before title. in mobile view?**
> 
> Screenshot 2022-09-23 at 09 54 58
> 
> Screenshot 2022-09-23 at 09 55 06

What do you mean by the unwanted space? If you're talking about the fact that the title is centered, it's supposed to be that way :).

File: src/components/FarmingList/ListItem.jsx (hunk 1 above)
Comment (guerrap) [reply src/components/FarmingList/ListItem.jsx:82]: Yes, we can use `carrotListLogoUrl` directly

**Comment by guerrap** [comment]
Improve the farm's details page, by adding more information about the campaign:

image

**Pull request merged**